File: sdk/samples/all in on code/Visual Studio 2010/CSTFSEventListener/IEventService.cs
/****************************** Module Header ******************************\
* Module Name:  IEventService.cs
* Project:	    CSTFSEventListener
* Copyright (c) Microsoft Corporation.
* 
* Define the interface that all TFS event listener must implement. 
* 
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System.ServiceModel;

namespace CSTFSEventListener
{
    [ServiceContract(Namespace = "http://schemas.microsoft.com/TeamFoundation/2005/06/Services/Notification/03")]
    public interface IEventService
    {
        [OperationContract(Action = "http://schemas.microsoft.com/TeamFoundation/2005/06/Services/Notification/03/Notify")]
        [XmlSerializerFormat(Style = OperationFormatStyle.Document)]
        void Notify(string eventXml, string tfsIdentityXml);
    }
}

File: sdk/samples/all in on code/Visual Studio 2010/VBTFSEventListener/IEventService.vb
'*************************** Module Header ******************************'
' Module Name:  IEventService.vb
' Project:	    VBTFSEventListener
' Copyright (c) Microsoft Corporation.
' 
' Define the interface that all TFS event listener must implement. 
' 
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'*************************************************************************'

Imports System.ServiceModel

<ServiceContract(Namespace:="http://schemas.microsoft.com/TeamFoundation/2005/06/Services/Notification/03")>
Public Interface IEventService
    <OperationContract(Action:="http://schemas.microsoft.com/TeamFoundation/2005/06/Services/Notification/03/Notify")>
    <XmlSerializerFormat(Style:=OperationFormatStyle.Document)>
    Sub Notify(ByVal eventXml As String, ByVal tfsIdentityXml As String)
End Interface
